presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/raw/RawRowEncoder.java

```diff
@@ -0,0 +1,290 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kafka.encoder.raw;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableSet;
+import io.prestosql.plugin.kafka.encoder.AbstractRowEncoder;
+import io.prestosql.plugin.kafka.encoder.EncoderColumnHandle;
+import io.prestosql.spi.connector.ConnectorSession;
+import io.prestosql.spi.type.Type;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.OptionalInt;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static io.prestosql.spi.type.BooleanType.BOOLEAN;
+import static io.prestosql.spi.type.DoubleType.DOUBLE;
+import static io.prestosql.spi.type.IntegerType.INTEGER;
+import static io.prestosql.spi.type.RealType.REAL;
+import static io.prestosql.spi.type.SmallintType.SMALLINT;
+import static io.prestosql.spi.type.TinyintType.TINYINT;
+import static io.prestosql.spi.type.Varchars.isVarcharType;
+import static java.lang.Integer.parseInt;
+import static java.lang.String.format;
+
+public class RawRowEncoder
+        extends AbstractRowEncoder
+{
+    private enum FieldType
+    {
+        BYTE(Byte.SIZE),
+        SHORT(Short.SIZE),
+        INT(Integer.SIZE),
+        LONG(Long.SIZE),
+        FLOAT(Float.SIZE),
+        DOUBLE(Double.SIZE);
+
+        private final int size;
+
+        FieldType(int bitSize)
+        {
+            this.size = bitSize / 8;
+        }
+
+        public int getSize()
+        {
+            return size;
+        }
+    }
+
+    private static final Pattern MAPPING_PATTERN = Pattern.compile("(\\d+)(?::(\\d+))?");
+    private static final Set<Type> SUPPORTED_PRIMITIVE_TYPES = ImmutableSet.of(
+            BIGINT, INTEGER, SMALLINT, TINYINT, DOUBLE, REAL, BOOLEAN);
+
+    public static final String NAME = "raw";
+
+    private final ByteBuffer buffer;
+    private int[] valueLengths;
+    private int currentBufferPosition;
+
+    public RawRowEncoder(ConnectorSession session, List<EncoderColumnHandle> columnHandles)
+    {
+        super(session, columnHandles);
+
+        int capacity = 0;
+        for (int i = 0; i < columnHandles.size(); i++) {
+            capacity += valueLengths[i];
+        }
+
+        this.buffer = ByteBuffer.allocate(capacity);
+        this.currentBufferPosition = 0;
+    }
+
+    @Override
+    protected void validateColumns(List<EncoderColumnHandle> columnHandles)
```

Comment:
I have second thoughts why do we need `validateColumns` at all. We do the validation in the constructor of specific `RowEncoder`

Reply:
that's a good point, how should I go about changing that? Should I just change it in each of the encoder prs or would it be better to change it from a different pr?

Reply:
Different small refactoring PR would be be better. We can merge it quickly.

Reply:
@aalbu do  you agree with change?

Reply:
Sorry, I missed this, but yes.